test/search_api/filters/test_search_api_include_filter.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+
+from datagateway_api.src.datagateway_api.filter_order_handler import FilterOrderHandler
+from datagateway_api.src.search_api.filters import SearchAPIIncludeFilter
+from datagateway_api.src.search_api.query import SearchAPIQuery
+
+
+class TestSearchAPIIncludeFilter:
+    def test_valid_apply_filter_single(self):
+        filter_input = SearchAPIIncludeFilter("documents", "Dataset")
+        entity_name = "Dataset"
+        expected_query = "SELECT o FROM Dataset o INCLUDE o.investigation"
+
+        filter_handler = FilterOrderHandler()
+        filter_handler.add_filter(filter_input)
+        test_query = SearchAPIQuery(entity_name)
+
+        filter_handler.apply_filters(test_query)
+
+        assert expected_query == str(test_query.icat_query.query)
+
+    @pytest.mark.parametrize(
+        "test_include_filter, entity_name, expected_query",
+        [
+            pytest.param(
+                SearchAPIIncludeFilter(["documents"], "Dataset"),
+                "Dataset",
+                "SELECT o FROM Dataset o INCLUDE o.investigation",
+                id="Dataset.documents",
+            ),
+            pytest.param(
+                SearchAPIIncludeFilter(["datasets"], "Document"),
+                "Document",
+                "SELECT o FROM Investigation o INCLUDE o.datasets",
+                id="Document.datasets",
+            ),
+            pytest.param(
+                SearchAPIIncludeFilter(["samples"], "Dataset"),
+                "Dataset",
+                "SELECT o FROM Dataset o INCLUDE o.sample",
+                id="Dataset.samples",
+            ),
+            pytest.param(
+                SearchAPIIncludeFilter(["members"], "Document"),
+                "Document",
+                "SELECT o FROM Investigation o INCLUDE o.investigationUsers",
+                id="Document.members",
+            ),
+            pytest.param(
+                SearchAPIIncludeFilter(["members.person"], "Document"),
+                "Document",
+                "SELECT o FROM Investigation o INCLUDE o.investigationUsers AS iu,"
+                " iu.user",
+                id="Document.members.person",
+            ),
+            # TODO - Below test fails because dataset/investigation parameters issue on
+            # mapping
+            pytest.param(
+                SearchAPIIncludeFilter(["dataset.parameters.document"], "File"),
+                "File",
+                "SELECT o FROM Datafile o INCLUDE o.dataset AS d, d.parameters AS p,"
+                " p.investigation",
+                id="File.dataset.parameters.document",
+                marks=pytest.mark.skip,
+            ),
```

Comment:
Like we discussed, a `DatasetParameter` entity does not have an `investigation` field so this test can be removed.
```suggestion
```

Reply:
Agreed, just removed it